compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BTypeReferenceType.java

```diff
@@ -61,6 +61,6 @@ public TypeKind getKind() {
 
     @Override
     public boolean isNullable() {
-        return this.referredType.isNullable();
+        return this.referredType != null && this.referredType.isNullable();
```

Comment:
Won't need this with the changes in https://github.com/ballerina-platform/ballerina-lang/pull/34137.